src/openforms/js/components/admin/form_design/useConfirm.js

```diff
@@ -0,0 +1,57 @@
+import {useState} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import ActionButton from 'components/admin/forms/ActionButton';
+import {Modal} from 'components/admin/modals';
+
+const useConfirm = (message, title = '') => {
+  const [promise, setPromise] = useState(null);
+
+  const confirm = () =>
+    new Promise(resolve => {
+      setPromise({resolve});
+    });
+
+  const handleClose = () => {
+    setPromise(null);
+  };
+
+  const handleConfirm = () => {
+    promise?.resolve(true);
+    handleClose();
+  };
+
+  const handleCancel = () => {
+    promise?.resolve(false);
+    handleClose();
+  };
+
+  const ConfirmationModal = () => {
+    const intl = useIntl();
+    const confirmBtnText = intl.formatMessage({
+      description: 'Confirmation modal confirm button',
+      defaultMessage: 'Confirm',
+    });
+    const cancelBtnText = intl.formatMessage({
+      description: 'Confirmation modal cancel button',
+      defaultMessage: 'Cancel',
+    });
+    return (
+      <Modal
+        title={title}
+        isOpen={promise !== null}
+        contentModifiers={['with-form', 'small']}
+        closeModal={handleCancel}
+      >
+        <div style={{flexGrow: 1}}>{message}</div>
```

Comment:
Maybe a class for this should be added where this styling is set instead?

Reply:
With some modal styling changes is this inline styling no longer needed 👍 

Extra info: https://github.com/open-formulieren/open-forms/pull/4814#discussion_r1838513591